File: leetsync/2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.java
class Solution {
    public int[] vowelStrings(String[] words, int[][] queries) {
        int wordsLen = words.length;
        int[] prefixArray = new int[wordsLen];
        int sum = 0;
        for (int i = 0; i < wordsLen; i++) {
            String word = words[i];
            if (isWordStartEndWithVowel(word)) {
                sum++;
            }
            prefixArray[i] = sum;
        }
        int queriesLen = queries.length;
        int[] result = new int[queriesLen];
        for (int i = 0; i < queriesLen; i++) {
            int l = queries[i][0];
            int r = queries[i][1];
            result[i] = prefixArray[r] - (l > 0 ? prefixArray[l - 1] : 0);
        }
        return result;
    }

    public boolean isWordStartEndWithVowel(String word) {
        char first = word.charAt(0);
        char last = word.charAt(word.length() - 1);
        return isVowel(first) && isVowel(last);
    }

    private boolean isVowel(char character) {
        return character == 'a' || character == 'e' || character == 'i' || character == 'o' || character == 'u';
    }
}

File: leetsync/2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.go
func vowelStrings(words []string, queries [][]int) []int {
	wordsLen := len(words)
	prefixArray := make([]int, wordsLen)
	sum := 0
	for i := 0; i < wordsLen; i++ {
		word := words[i]
		if isWordStartEndWithVowel(word) {
			sum++
		}
		prefixArray[i] = sum
	}
	queriesLen := len(queries)
	result := make([]int, queriesLen)
	for i := 0; i < queriesLen; i++ {
		l := queries[i][0]
		r := queries[i][1]
		left := 0
		if l > 0 {
			left = prefixArray[l-1]
		}
		result[i] = prefixArray[r] - left
	}
	return result
}
func isWordStartEndWithVowel(word string) bool {
	first := word[0]
	last := word[len(word)-1]
	return isVowel(first) && isVowel(last)
}

func isVowel(ch byte) bool {
	return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u'
}